dart-impl/base/src/locality.c

```diff
@@ -987,7 +1051,7 @@ dart_ret_t dart__base__locality__group_subdomains(
   group_domain->num_units = 0;
   group_domain->num_nodes = 0;
   for (int gd = 0; gd < num_grouped; gd++) {
-    memcpy(&group_domain->domains[gd], &grouped_domains[gd],
+    memcpy(group_domain->children[gd], &grouped_domains[gd],
```

Comment:
Mmmmmh. Even worse: I changed `dart_domain_locality_t.children` from array-of-domains to array-of-domain-pointers to make adding / moving / removing of sub-domains more efficient.
The grouped domain shoud not be mem-copied here, only it's address.